device/core/test/_internal_client_test.js

```diff
@@ -780,6 +780,120 @@ var ModuleClient = require('../dist/module_client').ModuleClient;
         });
       });
     });
+
+    describe('sendTelemetry', function () {
+      [42, {}, true].forEach(function (val) {
+        it(`throws if the second parameter is neither a function nor a string (${typeof val})`, function () {
+          var client = new ClientCtor(new EventEmitter());
+          assert.throws(function () {
+            client.sendTelemetry({temperature: '42'}, val);
+          }, TypeError, 'The second parameter to sendTelemetry must be a function (sendTelemetryCallback) or string (componentName)');
+        });
+      });
+
+      [42, {}, true, 'string'].forEach(function (val) {
+        it(`throws if the third parameter is not a function (${typeof val})`, function () {
+          var client = new ClientCtor(new EventEmitter());
+          assert.throws(function () {
+            client.sendTelemetry({temperature: '42'}, "thermostat1", val);
+          }, TypeError, 'Callback has to be a Function');
+        });
+      });
+
+      it('correctly populates a Message, calls sendEvent, and returns a Promise if no component or callback is specified', function (testCallback) {
+        var payload = {temperature: 42};
+        var client = new ClientCtor(new EventEmitter());
+        client.sendEvent = (message, callback) => {
+          try {
+            assert.strictEqual(message.contentEncoding, 'utf-8');
+            assert.strictEqual(message.contentType, 'application/json');
+            assert.strictEqual(message.data, JSON.stringify(payload));
+            assert.isUndefined(message.properties.getValue('$.sub'));
+            assert.isUndefined(callback);
+          } catch (err) {
+            return Promise.reject(err);
+          }
+          return Promise.resolve(new results.MessageEnqueued());
+        }
+        client.sendTelemetry(payload).then(
+          function () {
+            testCallback();
+          },
```

Comment:
Isn't the normal pattern to have a .catch in the error case?

Reply:
.catch is just syntactic sugar for the two argument .then